src/Data/Bool/Base.agda

```diff
@@ -73,3 +73,7 @@ infix 0 if_then_else_
 if_then_else_ : Bool → A → A → A
 if true  then t else f = t
 if false then t else f = f
+
+strictify : Bool → Bool
+strictify b = if b then true else false
```

Comment:
Oh my. This is *the* code that I consider amongst the most evil! Can't it at least be written
```agda
strictify true = true
strictify false = false
```
?

Reply:
Agree that, in novice programmers, that one-line expression is truly evil. 

But it nevertheless is precisely equivalent to your definition (up to reduction/strictness behaviour), for the (additional) cost of a (generatively distinct) function definition. 

Cf. defining `_++_` directly, or as an instance of `foldr`?

Reply:
I sure hope those definitions are precisely equivalent! `strictify` is weird, I don't really mind if it has a cost.

I personally think the computational `_++_` should be defined in terms of `foldr`.

Reply:
Well, as with my comments below, I *believe* on my understanding of Agda's evaluation strategy that they are reduction-behaviour equivalent, in that they are each
* blocked, neutral terms, if the `Bool` scrutinee is unevaluated/neutral
* the relevant literal value, depending on the value of the scrutinee

I've wondered if, as in a separate issue/PR, we considered systematically adding the dependent/non-dependent eliminators for `Data.X`, the 'evil' arises because of our habit of *not* seeing `if_then_else` as 'merely' the non-dependent eliminator for `Bool`? (The separate evil of novice programmers writing `if b then true else false` in a strict language is another matter...)